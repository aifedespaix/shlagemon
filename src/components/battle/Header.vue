<script setup lang="ts">
import type { Trainer } from '~/type'

const props = defineProps<{ zoneName?: string, trainer?: Trainer, defeated?: number }>()
const defeated = computed(() => props.defeated ?? 0)
const zone = useZoneStore()
const info = useZoneInfoStore()
</script>

<template>
  <div class="w-full flex items-center gap-2 overflow-hidden font-bold" :class="props.trainer ? 'justify-end' : 'justify-center'">
    <template v-if="props.zoneName && info.hasMultipleZones">
      <div class="flex flex-col items-center justify-center p-x-20">
        <div class="w-full overflow-hidden text-ellipsis whitespace-nowrap text-center" :title="props.zoneName">
          {{ props.zoneName }}
        </div>
        <div v-if="zone.current.type === 'sauvage'" class="whitespace-nowrap text-xs">
          {{ zone.current.minLevel }} - {{ zone.current.maxLevel }}
        </div>
      </div>
    </template>

    <template v-else-if="props.trainer">
      <div class="flex items-stretch gap-2">
        <div class="flex flex-col items-end justify-between">
          <div>{{ props.trainer.character.name }}</div>
          <div class="flex gap-2">
            <UiImageByBackground
              v-for="i in props.trainer.shlagemons.length"
              :key="i"
              src="/items/shlageball/shlageball.webp"
              class="h-4 w-4"
              :class="{ 'saturate-0': i <= defeated }"
            />
          </div>
        </div>

        <!-- Le point clÃ© : [contain:size] + stretch -->
        <div class="[contain:size] w-8 flex shrink-0 items-center self-stretch justify-center overflow-hidden">
          <CharacterImage
            :id="props.trainer.character.id"
            :alt="props.trainer.character.name"
            class="max-h-full max-w-full object-contain"
          />
        </div>
      </div>
    </template>
  </div>
</template>
