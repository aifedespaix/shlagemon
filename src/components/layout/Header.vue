<script setup lang="ts">
// future header logic
</script>

<template>
  <header class="flex items-center justify-between bg-gray-100 p-4 dark:bg-gray-800">
    <img src="/logo.png" alt="Logo ShlagÃ©mon" class="h-8">
    <div class="flex gap-2">
      <!-- TODO: theme toggle -->
    </div>
  </header>
</template>
