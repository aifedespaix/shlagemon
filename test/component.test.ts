import { mount } from '@vue/test-utils'
import { createPinia, setActivePinia } from 'pinia'
import { describe, expect, it } from 'vitest'
import Header from '../src/components/layout/Header.vue'

describe('component Header.vue', () => {
  it('should render', () => {
    const pinia = createPinia()
    setActivePinia(pinia)
    const wrapper = mount(Header, {
      global: { plugins: [pinia] },
    })
    expect(wrapper.find('header').exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
