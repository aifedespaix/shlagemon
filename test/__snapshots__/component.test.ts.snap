// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`component Header.vue > should render 1`] = `
"<header class="h-12 flex items-center justify-between bg-gray-100 p-4 dark:bg-gray-800"><img src="/logo.png" alt="Logo Shlagémon" class="h-20 -my-4">
  <div class="flex items-center gap-2"><button aria-label="Changer de thème" class="relative h-6 w-12 rounded-full bg-gray-200 transition-colors dark:bg-gray-700" type="button"><span class="absolute left-0.5 top-0.5 h-5 w-5 flex items-center justify-center rounded-full bg-white text-yellow-500 transition-all dark:bg-gray-900 dark:text-yellow-300"><div class="i-carbon-sun"></div></span></button><button class="p-1" aria-label="Reset Save">
      <div class="i-carbon-trash-can text-red"></div>
    </button></div>
</header>"
`;
